C#/singleton/Logger.cs

```diff
@@ -1,10 +1,70 @@
-namespace Singleton
+namespace Singleton;
+
+/// <summary>
+/// The Singleton class defines the <see cref="GetInstance"/> method that serves as an
+/// alternative to constructor and lets clients access the same instance of
+/// this class over and over.
+///
+/// The Singleton should always be a 'sealed' class to prevent class
+/// inheritance through external classes and also through nested classes.
+/// </summary>
+public sealed class Logger
 {
-    public enum LogLevel
+    private static Logger? s_instance;
+    private LogLevel _logLevel = LogLevel.INFO;
+    private readonly static object LockObject = new();
```

Comment:
I can't remember exactly the convention for writing read-only fields, but shouldn't that be the same as for all private members? 

```C#
private readonly static object _lockObject = new();
```

Reply:
@YoussefWaelMohamedLotfy here it's

Reply:
Did it appear now? @YoussefWaelMohamedLotfy  🏃‍♂️

Reply:
@mahyoussef this is from VS IDE Rules
<img width="586" alt="image" src="https://github.com/mahyoussef/ultimate-design-patterns/assets/40206862/88169d44-8d27-4a9d-8c0e-0acd5927aac5">

<img width="360" alt="image" src="https://github.com/mahyoussef/ultimate-design-patterns/assets/40206862/5faf927d-c147-472f-8b40-e26af0f27732">

Reviewed the suggestion and just applied it.